### PR TITLE
Merging in the Base Level Code for Super Fast Blink

### PR DESCRIPTION
**Pull request by jmacgregor-accenture**
Based on part one of this tutorial

### CODE REVIEW

File: timer-blink/timed-blink.c [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        DDRB |= (1 << DDB5);
+    }
+    else if (bit == 4)
+    {
+        DDRB |= (1 << DDB4);
+    }
+    else if (bit == 2)
+    {
+        DDRB |= (1 << DDB4);
```
Comment (rubberduck203) [inline timer-blink/timed-blink.c]: Wrong bit

File: timer-blink/timed-blink.c [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        PORTB &= ~(1 << PORTB5);
+    }
+    else if (bit == 4)
+    {
+        PORTB &= ~(1 << PORTB4);
+    }
+    else if (bit == 2)
+    {
+        PORTB &= ~(1 << PORTB2);
```
Comment (rubberduck203) [inline timer-blink/timed-blink.c]: If you passed `PORTBx` directly, you could remove the overhead of the `if..else if`

File: timer-blink/timed-blink.c [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        PORTB |= (1 << PORTB2);
+    }
+
+    light_on = true;
+}
+
+void toggle_led(int bit)
+{
+    if (light_on)
```
Comment (rubberduck203) [inline timer-blink/timed-blink.c]: You can toggle with an `xor` operation.

File: timer-blink/timed-blink.c [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        switch_power_on(bit);
+    }
+    
+}
+
+int main(void)
+{
+    setup_bit_as_output(5);
+
+    TCCR1B |= (1 << CS10); // How do we know to use this clock source??
```
Comment (rubberduck203) [inline timer-blink/timed-blink.c]: Let’s look at the datasheet together.

File: timer-blink/timed-blink.c (hunk 3 above)
Comment (jmacgregor-accenture) [reply timer-blink/timed-blink.c]: Refactored to do just that.

File: timer-blink/timed-blink.c (hunk 2 above)
Comment (jmacgregor-accenture) [reply timer-blink/timed-blink.c]: Definitely. Removed the copypasta from another program to make this code fit the problem.

File: timer-blink/timed-blink.c (hunk 1 above)
Comment (jmacgregor-accenture) [reply timer-blink/timed-blink.c]: Good catch--killed the whole branching logic anyway.

**Pull request merged**